spec/support/shared_examples/index_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.shared_examples 'index examples' do |class_name, resource_name, path_name|
+  let(:user)  { create(:user) }
+  let!(:list) { FactoryBot.create_list(resource_name, 10) }
+
+  before { user.confirm }
+
+  context 'valid request' do
+    before { get path_name, headers: headers_aux(user), as: :json }
+
+    it 'returns status 200 OK' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'returns resources' do
+      expect(json).not_to be_empty
+      expect(json[class_name.to_s].size).to eq(list.size)
```

Comment:
It's always better to have one expect per spec (`it`). In this case, this spec is included in the last one, in case both array matches, that means they have the same size.